app/assets/javascripts/student_profile/record_service.js

```diff
@@ -78,7 +78,8 @@
     },
 
     onProvidedByEducatorTyping: function(event) {
-      this.setState({ providedByEducatorName: event.target.value });
+      var updatedName = (event.target.value != "") ? event.target.value : null;
```

Comment:
When looking at #470 I noticed that we only initialise providedByEducatorName when the user starts typing. Thus a null value meant nothing was entered and we could say "No staff recorded". You're right that an empty string would be the better choice. Fixes the original issue without me having to set the thing back to null if it's blank again. Guess I was working around the issue instead of fixing it!